src/providers/pdal/qgspdalprovider.cpp

```diff
@@ -376,8 +377,10 @@ QString QgsPdalProviderMetadata::filters( QgsProviderMetadata::FilterType type )
       return QString();
 
     case QgsProviderMetadata::FilterType::FilterPointCloud:
-      // TODO get the available/supported filters from PDAL library
-      return QObject::tr( "PDAL Point Clouds" ) + QStringLiteral( " (*.laz *.las *.LAZ *.LAS)" );
+      if ( !sSupportedFormatsPopulated )
+        buildSupportedPointCloudFileFilterAndExtensions();
```

Comment:
what happens if querySublayers/priorityForUri is called before `filters`?

I'd suggest calling buildSupportedPointCloudFileFilterAndExtensions from both those methods too, and using a standard std::once guard instead of sSupportedFormatsPopulated

Reply:
I could also call it from the constructor, but `filters` is called by `QgsProviderRegistry::init()` during startup so it looks safe